iip-Smart_Treasury.md

```diff
@@ -0,0 +1,93 @@
+
+## Abstract
+
+**Issue**: Looking for a function to accrue value to INDEX tokens that is gas/tax efficient and allows the Index Coop to retain income for reinvestment during the growth phase.
+
+**Solution**: Deploy a smart balancer pool for a portion of our treasury as an automated buy-back mechanism for the INDEX token.
+
+**Why**:
+* Streaming fees accrue value directly to INDEX tokens via the automated buy-back mechanism
+* Maintains a fixed amount of treasury diversification because of fixed pool weight
+* Sustainability in terms of being able to reward contributors from the pool into the future
+* Increased market liquidity
+* Improved treasury transparency
+* Traders pay us to rebalance the pool by arbitraging the price difference after a deposit/withdrawal. [Further reading](https://balancer.finance/whitepaper/)
+
+**How**: 80/20 INDEX/ETH Balancer smart pool
+
+**Immediate Effect**: $100k DPI swapped to ETH, combine with $400k INDEX from treasury committee wallet and seed the pool. Manually sell DPI streaming fees to ETH and deposit into the pool over the following 3 months.
+
+Implementing this proposal would involve the creation of a balancer smart pool that only a whitelisted controller contract (in this case the Index Coop multi-sig) can interact with. We will deploy the pool through Balancer’s GUI using the parameters set out in the Specification section below. This will provide enough funds to cover 6 months of reward contributions plus sufficient liquidity to encourage arbitrage trades.
+
+We will be able to monitor the performance of the pool and determine slippage, trading fees and Balancer rewards. The data can be used to project if this is worthwhile scaling up and adopting on a permanent basis, as well as how to best adjust the trading fee parameter for optimal growth.
+
+## Specification - Initial
+
+Weight: 80/20 INDEX/ETH
+
+Liquidity: $500k seeding ($400k INDEX, 100k ETH), further liquidity added via DPI converted to ETH
+
+Swap fee: 0.3% (can be changed)
+
+Cost: Negligible, gas fees are reimbursed for deployment
+
+Scope of work: deploy via GUI, manually sell streaming fees for ETH and deposit in the pool. 
+
+Configurable rights: Key - **True**, False
+
+**canPauseSwapping**
+
+**canChangeSwapFee**
+
+**canChangeWeights**
+
+**canAddRemoveTokens**
+
+**canWhitelistLPs**
+
+canChangeCap - false, allowing us the flexibility to add any amount of liquidity
+
+## Specification - >3 months
+
+Weight: 80/20 INDEX/ETH
+
+Liquidity: $1.5m, with plans to add further liquidity via all live product streaming fees
+
+Swap fee: 0.3% (can be changed)
+
+Cost: Development (minimal), contract spec below
+
+Scope of work: Develop a contract to convert streaming fees to ETH and deposit into smart treasury (shown in red box below).  
+
+![image](https://github.com/darkforestcap/IIPS/blob/master/streaming_fee_conversion.png)
```

Comment:
Put your image in the `/assets` folder of this repo and add the link here.
Example in [the metaverse PR](https://github.com/SetProtocol/IIPS/pull/25/files#diff-d8d645d8b7c2dd2e210095bdff325617989c299300fb8582fe57e5492e75e778R95)